x-pack/filebeat/input/awss3/config.go

```diff
@@ -9,18 +9,26 @@ import (
 	"regexp"
 	"time"
 
+	"github.com/dustin/go-humanize"
+	"github.com/elastic/beats/v7/libbeat/reader/multiline"
+	"github.com/elastic/beats/v7/libbeat/reader/readfile"
 	awscommon "github.com/elastic/beats/v7/x-pack/libbeat/common/aws"
 )
 
 type config struct {
-	APITimeout               time.Duration       `config:"api_timeout"`
-	ExpandEventListFromField string              `config:"expand_event_list_from_field"`
-	FileSelectors            []FileSelectorCfg   `config:"file_selectors"`
-	FipsEnabled              bool                `config:"fips_enabled"`
-	MaxNumberOfMessages      int                 `config:"max_number_of_messages"`
-	QueueURL                 string              `config:"queue_url" validate:"nonzero,required"`
-	VisibilityTimeout        time.Duration       `config:"visibility_timeout"`
-	AwsConfig                awscommon.ConfigAWS `config:",inline"`
+	APITimeout               time.Duration           `config:"api_timeout"`
+	ExpandEventListFromField string                  `config:"expand_event_list_from_field"`
+	FileSelectors            []FileSelectorCfg       `config:"file_selectors"`
+	FipsEnabled              bool                    `config:"fips_enabled"`
+	MaxNumberOfMessages      int                     `config:"max_number_of_messages"`
+	QueueURL                 string                  `config:"queue_url" validate:"nonzero,required"`
+	VisibilityTimeout        time.Duration           `config:"visibility_timeout"`
+	AwsConfig                awscommon.ConfigAWS     `config:",inline"`
+	MaxBytes                 int                     `config:"max_bytes" validate:"min=0,nonzero"`
+	Multiline                *multiline.Config       `config:"multiline"`
```

Comment:
Do you think these options should be available in the `FileSelectorCfg`?

Reply:
Hmmmm  we could...

The use case would be a single S3 bucket that has a mix of multiline and non-multiline log files.

I'll add it, and we can see if we like it.

Reply:
The the parser options could be combined into a struct that is embedded in both `config` and `FileSelectorCfg` to avoid having to duplicate the same config struct tags.

Reply:
I'm thinking of changing this so you always have a file_selector, the default is just to match any filename.  That should make the config and code paths cleaner.  I know we want to get a build out with multiline "soon".  You OK with merging as is, with a new PR this week to clean this up?